pkg/util/hint/hint.go

```diff
@@ -312,6 +330,40 @@ func ParseStmtHints(hints []*ast.TableOptimizerHint,
 		case "straight_join":
 			hintOffs[hint.HintName.L] = i
 			straightJoinHintCnt++
+		case "hypo_index":
+			// to make it simpler, use Tables[0] as table, Tables[1] as index name, and Tables[2:] as column name.
+			if len(hint.Tables) < 3 {
+				warn := errors.NewNoStackErrorf("Invalid HYPO_INDEX hint, valid usage: HYPO_INDEX(tableName, indexName, cols...)")
+				warns = append(warns, warn)
+				continue
+			}
+			db := hint.Tables[0].DBName.L
+			if db == "" {
+				db = currentDB
+			}
+			tbl := hint.Tables[0].TableName
+			idx := hint.Tables[1].TableName
+			var colNames []model.CIStr
+			var cols []*model.IndexColumn
+			for i := 2; i < len(hint.Tables); i++ {
+				colNames = append(colNames, hint.Tables[i].TableName)
+				cols = append(cols, &model.IndexColumn{
+					Name:   hint.Tables[i].TableName,
+					Offset: i - 2,
```

Comment:
should it align to the real column offset in tableInfo？

Reply:
No, the offset here means its offset in the index.